lib/ignored-paths.js

```diff
@@ -21,55 +22,139 @@ debug = debug("eslint:ignored-paths");
-            ignorePatterns = fs.readFileSync(filepath, "utf8").split(/\r?\n/).filter(nonEmpty);
-        } catch (e) {
-            e.message = "Cannot read ignore file: " + filepath + "\nError: " + e.message;
-            throw e;
-        }
+    return ignoreFileFinder.findInDirectoryOrParents(cwd);
+}
+
+/**
+ * Replace Windows with Unix style paths and remove ./ prefix
+ * @param {string} filepath Path to normalize
+ * @returns {string} Normalized filepath
+ */
+function normalizeFilepath(filepath) {
+    filepath = filepath.replace("\\", "/");
+    filepath = filepath.replace(/^\.\//, "");
+    return filepath;
+}
+
+/**
+ * Remove a prefix from a filepath
+ * @param {string} filepath Path to remove the prefix from
+ * @param {string} prefix Prefix to remove from filepath
+ * @returns {string} Normalized filepath
+ */
+function removePrefixFromFilepath(filepath, prefix) {
+    prefix += "/";
+    if (filepath.indexOf(prefix) === 0) {
+        filepath = filepath.substr(prefix.length);
     }
+    return filepath;
+}
 
-    return ["node_modules/**", "bower_components/**"].concat(ignorePatterns);
+/**
+ * Resolves a filepath
+ * @param {string} filepath Path resolve
+ * @param {string} baseDir Base directory to resolve the filepath from
+ * @returns {string} Resolved filepath
+ */
+function resolveFilepath(filepath, baseDir) {
+    if (baseDir) {
+        var base = path.resolve(baseDir);
+        filepath = removePrefixFromFilepath(filepath, base);
+        filepath = removePrefixFromFilepath(filepath, fs.realpathSync(base));
+    }
+    filepath.replace(/^\//, "");
+    return filepath;
 }
 
-var ignoreFileFinder;
+/**
+ * Normalize and resolve a filepath relative to a given base directory
+ * @param {string} filepath Path resolve
+ * @param {string} baseDir Base directory to resolve the filepath from
+ * @returns {string} Normalized and resolved filepath
+ */
+function normalizeAndResolveFilepath(filepath, baseDir) {
+    filepath = normalizeFilepath(filepath);
+    return resolveFilepath(filepath, baseDir);
+}
 
 /**
- * Find an ignore file in the current directory or a parent directory.
- * @param {string} cwd Current working directory
- * @returns {string} Path of ignore file or an empty string.
+ * Merge options with defaults
+ * @param {object} options Options to merge with DEFAULT_OPTIONS constant
+ * @returns {object} Merged options
  */
-function findIgnoreFile(cwd) {
-    if (!ignoreFileFinder) {
-        ignoreFileFinder = new FileFinder(ESLINT_IGNORE_FILENAME, cwd);
+function mergeDefaultOptions(options) {
+    options = (options || {});
+    Object.keys(DEFAULT_OPTIONS).forEach(function(key) {
+        if (typeof (options[key]) === "undefined") {
+            options[key] = DEFAULT_OPTIONS[key];
+        }
+    });
+    return options;
+}
+
+/**
+ * Merge options with defaults
+ * @param {string} message Hint to print with the compatibility warning
+ * @returns {void}
+ */
+function printOsCompatibilityWarning(message) {
```

Comment:
Function for handling the warning now passed as option `onOsCompatibilityIssue` to IgnorePaths, so that the decision to print a warning is made by `CLIEngine`. Does that address your concerns properly?